src/ZArray.jl

```diff
@@ -11,12 +11,13 @@ const concurrent_io_tasks = Ref(50)
 getfillval(::Type{T}, t::String) where {T <: Number} = parse(T, t)
 getfillval(::Type{T}, t::Union{T,Nothing}) where {T} = t
 
-struct SenMissArray{T,N,V} <: AbstractArray{Union{T,Missing},N}
+struct SenMissArray{T,N} <: AbstractArray{Union{T,Missing},N}
   x::Array{T,N}
+  senval::T
 end
```

Comment:
I knwo this was a quite ad-hoc definition of SenMissArray, probably better to keep the sentinel value in a struct field instead of a type parameter. Now that the idea was taken up also here: https://github.com/JuliaData/SentinelArrays.jl it might be an option to just use that implementation. This should not stop this PR from being merged though.